recipes/at-spi2-core/all/conanfile.py

```diff
@@ -0,0 +1,90 @@
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class LibnameConan(ConanFile):
+    name = "at-spi2-core"
+    description = "It provides a Service Provider Interface for the Assistive Technologies available on the GNOME platform and a library against which applications can be linked"
+    topics = ("conan", "atk", "accessibility")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://gitlab.gnome.org/GNOME/at-spi2-core/"
+    license = "LGPL-2.1-or-later"
+    generators = "pkg_config"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_x11": [True, False],
+        }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_x11": False,
+        }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+        
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        # FIXME: this package is linux only?
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+    
+    def build_requirements(self):
+        self.build_requires("meson/0.54.2")
+        self.build_requires("pkgconf/1.7.3")
+    
+    def requirements(self):
+        self.requires("glib/2.67.0")
+        if self.options.with_x11:
+            self.requires("xorg/system")
+        self.requires("dbus/1.12.20")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_meson(self):
+        meson = Meson(self)
+        defs = {}
+        defs["introspection"] = "no"
+        defs["docs"] = "false"
+        defs["x11"] = "yes" if self.options.with_x11 else "no"
+        args=[]
+        args.append("--datadir=%s" % os.path.join(self.package_folder, "res"))
+        args.append("--localedir=%s" % os.path.join(self.package_folder, "res"))
+        args.append("--wrap-mode=nofallback")
+        meson.configure(defs=defs, build_folder=self._build_subfolder, source_folder=self._source_subfolder, pkg_config_paths=".", args=args)
+        return meson
+
+    def build(self):
+        meson = self._configure_meson()
+        meson.build()
+
+    def package(self):
```

Comment:
Do not really know if following files are needed for a Gnome library.

1. Service file `lib/systemd/user/at-spi-dbus-bus.service` - safe to drop
2. A lot of LC_MESSAGES resources (`res` directory)
3. Two binaries `bin/at-spi2-registryd`, `bin/at-spi2-registryd`

Reply:
Does it harm to keep these files in the package ? if the meson project installs them, someone could probably need them.

Reply:
It would not harm (at least I do not see a policy that denies packaging resources) but it is definitely not a minimal library. Meson can install them because `at-spi2-core` is not a pure library but also a localizable utility. If your other PRs based on `at-spi2` need those files, probably they should be kept.